src/c#/main/entity/entities/Pawn.cs

```diff
@@ -170,137 +145,66 @@ public class Pawn : Entity {
         }
 
-        private void gatherResources(Environment environment) {
-            if (!hasTargetEntity()) {
-                // select nearest tree or rock
+        // The current behavior type should only be changed in computeBehaviorType()
+        public void computeBehaviorType(Environment environment, NationRepository nationRepository) {
 
-                Entity nearestTree = environment.getNearestTree(getGameObject().transform.position);
-                Entity nearestRock = environment.getNearestRock(getGameObject().transform.position);
-                if (nearestTree == null && nearestRock == null) {
-                    return;
-                }
-                if (nearestTree == null) {
-                    setTargetEntity(nearestRock);
-                }
-                else if (nearestRock == null) {
-                    setTargetEntity(nearestTree);
-                }
-                else {
-                    float distanceToTree = Vector3.Distance(getGameObject().transform.position, nearestTree.getGameObject().transform.position);
-                    float distanceToRock = Vector3.Distance(getGameObject().transform.position, nearestRock.getGameObject().transform.position);
-                    if (distanceToTree < distanceToRock && getInventory().getNumItems(ItemType.WOOD) < targetNumWood) {
-                        setTargetEntity(nearestTree);
-                    }
-                    else {
-                        setTargetEntity(nearestRock);
+            if (energy < 80 && getInventory().getNumItems(ItemType.APPLE) == 0) {
+                // if nation leader has apples, purchase apples from leader
+                if (getNationId() != null) {
+                    Nation nation1 = nationRepository.getNation(getNationId());
+                    Entity nationLeader = environment.getEntity(nation1.getLeaderId());
+                    if (nationLeader.getInventory().getNumItems(ItemType.APPLE) > 0 && getInventory().getNumItems(ItemType.GOLD_COIN) >= 1) {
+                        currentBehaviorType = BehaviorType.PURCHASE_FOOD;
+                        return;
                     }
-                }
-            }
-            else if (isAtTargetEntity()) {
-                // gather
-                if (targetEntity.getType() == EntityType.TREE || targetEntity.getType() == EntityType.ROCK) {
-                    targetEntity.markForDeletion();
-                    getInventory().transferContentsOfInventory(targetEntity.getInventory());
-                    setTargetEntity(null);
-                }
-                else {
-                    Debug.LogWarning("target entity is not a tree or rock");
-                    setTargetEntity(null);
-                }
-            }
-            else {
-                moveTowardsTargetEntity();
-            }
-        }
+                } // TODO: fix this so this doesn't happen all the time
```

Comment:
Can you fix this before it's merged? It'll ensure that the git history has "working" code in it at all times on main.

Reply:
I'm not sure this is actually happening still, I will verify and resolve the problem if so!

Reply:
This is no longer happening.